internal/service/users.go

```diff
@@ -55,7 +56,19 @@ func (u *usersService) UpdateUserInfo(ctx context.Context, id string, input *Upd
 	return u.repo.Update(ctx, id, &upd)
 }
 
+func (i *SignupUserInput) Validate() error {
+	return validation.ValidateStruct(i,
+		validation.Field(&i.Email, validation.Required, is.Email),
```

Comment:
Eventually it makes sense to make password complexity check configurable (via parameters in Config), but let's keep this simple implementation for now

Reply:
Addressed in the latest commit.